app/assets/javascripts/discourse/app/controllers/insert-hyperlink.js

```diff
@@ -5,6 +5,12 @@ import ModalFunctionality from "discourse/mixins/modal-functionality";
 import { searchForTerm } from "discourse/lib/search";
 import { bind } from "discourse-common/utils/decorators";
 
+function prefixProtocol(url) {
+  return url.indexOf("://") === -1 && url.indexOf("mailto:") === -1
```

Comment:
Could I convince you to test for `url.indexOf("mailto:") !== 0` instead since "http://foo.com/mailto:something/" is a valid URL?

Reply:
I'm fine with this :+1: 

I decided to move this fn to lib:url and added more tests for this